Extensions/xkit_patches.js

```diff
@@ -828,48 +828,39 @@ XKit.extensions.xkit_patches = new Object({
 
 			/**
 			 * Get the secure_form_key through a request using the current form_key
-			 * @param {Function} callback invoked with `{errors: boolean, kitten: String}`
+			 * @param {Function} callback - invoked with `{errors: Boolean, kitten: String}`
+			 * @param {Boolean} retry_mode - whether to retry using a new form_key
 			 */
-			XKit.interface.kitty.get = async function(callback) {
-				var m_object = {
-					errors: false,
-					kitten: ''
-				};
-
-				var kitty_diff = (new Date()) - XKit.interface.kitty.store_time;
-
+			XKit.interface.kitty.get = async function(callback, retry_mode = false) {
 				if (XKit.interface.kitty.stored !== "") {
+					const kitty_diff = (new Date()) - XKit.interface.kitty.store_time;
 					if (kitty_diff <= XKit.interface.kitty.expire_time && kitty_diff > 0) {
-						m_object.kitten = XKit.interface.kitty.stored;
-						callback(m_object);
+						callback({errors: false, kitten: XKit.interface.kitty.stored});
 						return;
 					}
 				}
 
-				if (!XKit.interface.form_key()) {
+				if (!XKit.interface.form_key() || retry_mode) {
```

Comment:
instead of using retry_mode here, maybe we could just invalidate `form_key` below, where we invalidate kitty.stored?

Reply:
we'd need to keep `retry_mode` as an argument to prevent actual failures from retrying infinitely, but yes